vault/resource_managed_keys_test.go

```diff
@@ -0,0 +1,194 @@
+					}
+				},
+				Config: testManagedKeysConfig_basic(name0, name1),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "aws.#", "2"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "aws.*",
+						map[string]string{
+							consts.FieldName:         name0,
+							consts.FieldKeyBits:      "2048",
+							consts.FieldKeyType:      "RSA",
+							consts.FieldKMSKey:       "alias/test_identifier_string",
+							consts.FieldAWSAccessKey: "ASIAKBASDADA09090",
+							consts.FieldAWSSecretKey: "8C7THtrIigh2rPZQMbguugt8IUftWhMRCOBzbuyz",
+						},
+					),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "aws.*",
+						map[string]string{
+							consts.FieldName:         name1,
+							consts.FieldKeyBits:      "4096",
+							consts.FieldKeyType:      "RSA",
+							consts.FieldKMSKey:       "alias/test_identifier_string_2",
+							consts.FieldAWSAccessKey: "ASIAKBASDADA09090",
+							consts.FieldAWSSecretKey: "8C7THtrIigh2rPZQMbguugt8IUftWhMRCOBzbuyz",
+						},
+					),
+				),
+			},
+			// This test removes one of the managed keys from the set
+			// and also updates the name for the other remaining key
+			// Tests: ForceNew on Name + Deletion of removed keys
+			{
+				Config: testManagedKeysConfig_updated(name0),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "aws.#", "1"),
+					resource.TestCheckTypeSetElemNestedAttrs(resourceName, "aws.*",
+						map[string]string{
+							consts.FieldName:         name0,
+							consts.FieldKeyBits:      "4096",
+							consts.FieldKeyType:      "RSA",
+							consts.FieldKMSKey:       "alias/test_identifier_string_2",
+							consts.FieldAWSAccessKey: "ASIAKBASDADA09090",
+							consts.FieldAWSSecretKey: "8C7THtrIigh2rPZQMbguugt8IUftWhMRCOBzbuyz",
+						},
+					),
+				),
+			},
+			// test out-of-band changes and UUID update
+			{
+				PreConfig: func() {
+					// Delete previously configured managed key from Vault
+					client := testProvider.Meta().(*provider.ProviderMeta).GetClient()
+
+					p := getManagedKeysPath(kmsTypeAWS, name0)
+					_, err := client.Logical().Delete(p)
+					if err != nil {
+						t.Fatalf("could not delete managed key")
+					}
+
+					// Recreate w/ same name; forces UUID update out-of-band
+					data := map[string]interface{}{
+						consts.FieldAWSAccessKey: "ASIAKBASDADA09090",
+						consts.FieldAWSSecretKey: "8C7THtrIigh2rPZQMbguugt8IUftWhMRCOBzbuyz",
+						consts.FieldKeyBits:      "4096",
+						consts.FieldKeyType:      "RSA",
+						consts.FieldKMSKey:       "alias/test_identifier_string_2",
+					}
+
+					_, err = client.Logical().Write(p, data)
+					if err != nil {
+						t.Fatalf("could not create managed key to Vault")
+					}
+				},
+				Config: testManagedKeysConfig_updated(name0),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "aws.#", "1"),
```

Comment:
This doesn't really prove that the a delete/create operation took place. In the end you should end up with a new UUID that is different from the original created by the provider and the one created manually.

Reply:
Tracking this task in a separate Jira. Thanks for the callout!